feat: Initial integration of orb-ns CLI into the Name System's operator API

jsantell opened the pull request:
* Implements `orb-ns status`
* Implements `orb-ns records [(get $DID)|(put $RECORD)]`
* Implements `orb-ns peers [ls|(add $ADDR)]`
* Synchronizes orb-ns flags with the config file variant for bootstrap nodes
* Allow SocketAddr/Url/Multiaddr to be represented by SocketAddr/IpAddr/u16 Ports in orb-ns flags and configs.

cdata commented:
Recorded test flake for posterity, with commentary: https://github.com/subconsciousnetwork/noosphere/issues/167#issuecomment-1419528064

cdata commented on rust/noosphere-ns/src/bin/orb-ns/cli_address.rs:
```diff
@@ -0,0 +1,246 @@
+use serde::de;
+use std::str::FromStr;
+
+pub fn parse_cli_address<T: TryFrom<CLIAddress>>(input: &str) -> Result<T, String> {
```
Even if they are small, some member docs on these would be welcome

cdata commented on rust/noosphere-ns/src/bin/orb-ns/main.rs:
```diff
@@ -43,5 +55,39 @@ async fn main() -> Result<()> {
+        CLICommand::Status { api_url } => {
+            let client = HTTPClient::new(api_url).await?;
+            let info = client.network_info().await?;
+            println!("{:#?}", info);
```
Let's prefer `tracing::{info,debug,error,warn,trace}` to `println` throughout this change!

cdata reviewed (changes requested):
This looks great; some light cleanup is requested.

jsantell commented:
* Updated `println!` to `tracing::info!` note, these are a little sloppy at the moment -- future PR with full JSON responses for the "getter" cli commands (`orb-ns records get X`)
* Fixed module-docs, and added docs/doctests for some CLIAddress conversions

jsantell commented:
bin doctests are *not* run on `cargo test` -- looking into this

jsantell commented:
Ok updated and moved `CLIAddress` conversion tests to regular unit tests (so they run on `cargo run`), fixed some issues, and had to roll a de/serializersto handle both int and str types. Ready for one more pass!

cdata reviewed (approved):
Great work!
